### PR TITLE
DC-268 add policy actions for catalog admin

### PR DESCRIPTION
**Pull request by tcjiang9**
As part of setting up catalog SAM resources for this [ticket](https://broadworkbench.atlassian.net/jira/software/c/projects/DC/boards/118?modal=detail&selectedIssue=DC-268&assignee=604f833245a3bb0068e42e4c)

I need to be able to modify policies on our resource type catalog and its resource ids

### CODE REVIEW

File: src/main/resources/reference.conf [hunk 1]
```diff
@@ -871,6 +871,15 @@ resourceTypes = {
   }
   catalog = {
     actionPatterns = {
+      "share_policy::admin" = {
```
Comment (s-rubenstein) [inline src/main/resources/reference.conf]: Do we need this specifically? I think `alter_policies` covers a this, and this is just if you want certain users to be able to only share certain policies. Someone from CI might know better though?

Comment (tcjiang9) [reply src/main/resources/reference.conf]: you're right, we only need the admin group to be the admin anyways

**Review by pshapiro4broad — APPROVED** [review]
LGTM

**Review by fboulnois — APPROVED** [review]

**Review by ehigham — APPROVED** [review]

**Review by s-rubenstein — APPROVED** [review]
LGTM

**Comment by ehigham** [comment]
The integration test failure is knwon and caputured in https://broadworkbench.atlassian.net/browse/WOR-265

**Pull request merged**